index.js

```diff
@@ -5,99 +5,47 @@
 
 'use strict';
 
-var async = require('async'),
-	cheerio = require('cheerio'),
-	request = require('request'),
-	microdata = require('microdata-node');
+var BBPromise = require('bluebird');
+var cheerio = require('cheerio');
+var preq = require('preq');
+var microdata = require('microdata-node');
 
 // Default exported function
-exports = module.exports = function(urlOrOpts, callback){
-	request(urlOrOpts, function(error, response, html){
-		var chtml = cheerio.load(html);
-		exports.parseAll(chtml, function(err, results){
-			callback(err, results);
-		});
+exports = module.exports = function(urlOrOpts) {
+	return preq.get(urlOrOpts
+	).then(function(callRes) {
+		return BBPromise.resolve(exports.parseAll(cheerio.load(callRes.body)));
 	});
 };
 
 /**
- * Callback on Object containing all fields merged into
- * one object. The parameters key to a list which may contain
- * multiple values if multiples are found (for instance, if
- * multiple metadata types exist and both contain a parameter
- * called 'title')
- *
- * @param  {Object}   chtml     html Cheerio object to parse
- * @param  {Function} callback callback(error, mergedObject)
- */
-exports.parseAllMerged = function(chtml, callback){
-	var fcn, results, merged,
-		allMetadata = {},
-		metadataFunctions = exports.metadataFunctions;
-
-	async.eachSeries(Object.keys(metadataFunctions), function (key, cb){
-		fcn = metadataFunctions[key];
-		fcn(chtml, function(results){
-			if (results){
-				// Merge results into larger object
-				for (var key in results){
-					merged = allMetadata[key];
-					var value = results[key];
-
-					if (!merged){
-						merged = [];
-					}
-
-					if (value instanceof Array) {
-						merged = merged.concat(value);
-					} else {
-						merged.push(value);
-					}
-
-					allMetadata[key] = merged;
-				}
-			}
-		});
-		cb();
-	}, function(err) {
-		callback(err, allMetadata);
-	});
-};
-
-/**
- * Callback on Object containing all available datatypes, keyed
+ * Returns Object containing all available datatypes, keyed
  * using the same keys as in metadataFunctions.
  *
- * Currently only openGraph data as this is the only one implemented
- *
- * @param  {Object}   chtml     html Cheerio object to parse
- * @param  {Function} callback callback(error, allMetadata)
+ * @param  {Object}   chtml html Cheerio object to parse
+ * @return {Object}         contains metadata
  */
-exports.parseAll = function(chtml, callback){
-	var fcn,
-		allMetadata = {},
-		metadataFunctions = exports.metadataFunctions;
+exports.parseAll = function(chtml){
+	var fcn;
+	var meta;
+	var allMetadata = {};
+	var metadataFunctions = exports.metadataFunctions;
 
-	async.forEach(Object.keys(metadataFunctions), function (key, cb){
+	Object.keys(metadataFunctions).forEach(function(key) {
```

Comment:
Working on it in new WIP commit. (Just letting you know because in IRC you mentioned that you might do this :))